src/nemos/basis/_basis.py

```diff
@@ -835,33 +742,57 @@ class AdditiveBasis(Basis):
     """
 
     def __init__(self, basis1: Basis, basis2: Basis) -> None:
-        self.n_basis_funcs = basis1.n_basis_funcs + basis2.n_basis_funcs
-        super().__init__(self.n_basis_funcs, mode="eval")
+        CompositeBasisMixin.__init__(self, basis1, basis2)
+        Basis.__init__(self, mode="eval")
```

Comment:
why is this eval? shouldn't it be based on basis1 / basis2?

Reply:
we should probably say "composite", it is unspecified, one can be eval and one conv, or it could be eva/composite->eval/conv or any complex struct